### PR TITLE
Introduce gammapy.utils.fitting.Model base class

### PR DESCRIPTION
**Pull request by cdeil**
This PR introduces a `Model` base class. For now all it contains is `copy`, and it's not abstract because we don't have an abstract interface yet for all models that I could add.

@adonath - If you have any thoughts on something that should and can be added now already, let me know. Otherwise I think it's OK to introduce the base class now, and add to it piece by piece later.

Things that could go here are arithmetic operators, some helper methods to work with the parameters, expose the parameters as attributes, a registry that allows listing all model subclasses, ... In Astropy, they even managed to put `__call__` and `__init__` in the base class, and have a declarative scheme at the class-level for the parameters. Even that is possible if we change to a more uniform calling interface. All of those things would need to be prototyped before making a decision, and having the base class in place makes that easier (keep all existing tests working always).

@adonath - Please review.

### CODE REVIEW

File: gammapy/spectrum/models.py [hunk 1]
```diff
@@ -38,10 +37,6 @@ class SpectralModel(object):
     `~gammapy.spectrum.models.PowerLaw`.
     """
 
-    def __repr__(self):
```
Comment (adonath) [inline gammapy/spectrum/models.py]: Why did you remove this `__repr__`? I think it's useful to have it. I guess in the current implementation it could go to the `Model` base class as well?

Comment (cdeil) [reply gammapy/spectrum/models.py]: The current repr didn't show any useful information. IMO if we don't have any useful info beyond the class name that the default repr shows, we should just do nothing and show the default repr.

Yes, we probably should add `Model.__repr__` and have it show something like number of parameters or parameter names or ... it'll need some thought and discussion and once we agree on what would be useful to show we can add it back.

For that it'll help if this sub-class doesn't overwrite it and we don't notice, so I prefer to remove that now.
